winch/codegen/src/visitor.rs

```diff
@@ -637,6 +639,40 @@ where
         self.masm.store(typed_reg.reg.into(), addr, ty.into());
     }
 
+    fn visit_drop(&mut self) {
+        self.context.drop_last(1, |regalloc, val| match val {
+            Val::Reg(tr) => regalloc.free(tr.reg.into()),
+            Val::Memory(m) => self.masm.free_stack(m.slot.size),
+            _ => {}
+        });
+    }
+
+    fn visit_select(&mut self) {
+        let cond = self.context.pop_to_reg(self.masm, None);
+        let val2 = self.context.pop_to_reg(self.masm, None);
+        let val1 = self.context.pop_to_reg(self.masm, None);
+        let done = self.masm.get_label();
+        self.masm.branch(
```

Comment:
I'm somewhat surprised that select is being implemented with actual local control flow here; is there any reason we can't use a conditional move? That's the usual lowering in e.g. Cranelift (and avoiding actual branches is sometimes important at the guest-code level for Spectre mitigations).

Reply:
> is there any reason we can't use a conditional move? 

Good catch; no particular reason why we can't use a conditional move here, it's just that I didn't consider it initially, but there are very good arguments to do so. I've refactored this code to use conditional moves instead in https://github.com/bytecodealliance/wasmtime/pull/6912/commits/c56125cd833ff8107b285f4a2872d0930c014eef